File: music-domain/src/main/java/com/music/sale/domain/user/enum/UserRole.java
// Copyright (C) 2024 Your Name or Company
package com.music.sale.domain.user.enum;

public enum UserRole {
    ADMIN,
    SELLER,
    USER;

    public boolean hasPermission(UserRole requiredRole) {
        return switch (this) {
            case ADMIN -> true;
            case SELLER -> this == requiredRole;
            case USER -> this == requiredRole;
        };
    }
}

File: music-domain/src/main/kotlin/com/music/sale/domain/user/enum/UserRole.kt
// Copyright (C) 2024 Your Name or Company
package com.music.sale.domain.user.enum

enum class UserRole {
    ADMIN,
    SELLER,
    USER,
    ;

    fun hasPermission(requiredRole: UserRole): Boolean {
        return when (this) {
            ADMIN -> true
            SELLER -> this == requiredRole
            USER -> this == requiredRole
        }
    }
}
